pkg/v1/tkg/client/machine_deployment.go

```diff
@@ -54,18 +54,19 @@ type DeleteMachineDeploymentOptions struct {
 
 // NodePool a struct describing a node pool
 type NodePool struct {
-	Name             string                    `yaml:"name"`
-	Replicas         *int32                    `yaml:"replicas,omitempty"`
-	AZ               string                    `yaml:"az,omitempty"`
-	NodeMachineType  string                    `yaml:"nodeMachineType,omitempty"`
-	Labels           *map[string]string        `yaml:"labels,omitempty"`
-	VSphere          VSphereNodePool           `yaml:"vsphere,omitempty"`
-	Taints           *[]corev1.Taint           `yaml:"taints,omitempty"`
-	VMClass          string                    `yaml:"vmClass,omitempty"`
-	StorageClass     string                    `yaml:"storageClass,omitempty"`
-	Volumes          *[]tkgsv1alpha2.Volume    `yaml:"volumes,omitempty"`
-	TKR              tkgsv1alpha2.TKRReference `yaml:"tkr,omitempty"`
-	NodeDrainTimeout *metav1.Duration          `yaml:"nodeDrainTimeout,omitempty"`
+	Name                  string                    `yaml:"name"`
+	Replicas              *int32                    `yaml:"replicas,omitempty"`
+	AZ                    string                    `yaml:"az,omitempty"`
+	NodeMachineType       string                    `yaml:"nodeMachineType,omitempty"`
+	Labels                *map[string]string        `yaml:"labels,omitempty"`
+	VSphere               VSphereNodePool           `yaml:"vsphere,omitempty"`
+	Taints                *[]corev1.Taint           `yaml:"taints,omitempty"`
+	VMClass               string                    `yaml:"vmClass,omitempty"`
+	StorageClass          string                    `yaml:"storageClass,omitempty"`
+	Volumes               *[]tkgsv1alpha2.Volume    `yaml:"volumes,omitempty"`
+	TKR                   tkgsv1alpha2.TKRReference `yaml:"tkr,omitempty"`
+	NodeDrainTimeout      *metav1.Duration          `yaml:"nodeDrainTimeout,omitempty"`
+	MachineDeploymentBase string                    `yaml:"-"`
```

Comment:
Why `yaml:"-"`?

Reply:
So that it won't be deserialized if a user does include it in the config file for the node pool

Reply:
I don't think I get that atm. I would at least put a comment explaining why for future git archeologists.

Reply:
I don't see any harm if it can get picked up from the config file. As long as the command line param has priority, it's a-ok